pandas/core/ops/methods.py

```diff
@@ -97,7 +97,9 @@ def f(self, other):
             # this makes sure that we are aligned like the input
             # we are updating inplace so we want to ignore is_copy
             self._update_inplace(
-                result.reindex_like(self, copy=False)._data, verify_is_copy=False
+                result.reindex_like(self, copy=False)._data,
+                verify_is_copy=False,
+                change_cache=False,
```

Comment:
this is a band-aid; there must be something going wrong in _update_inplace itself.

Reply:
I took a different solution in the new commit. I've seen that the difference between inplace or not-inplace drop is that if we do it inplace, the "target" series keeps the weak reference to the original df. And thats why it later gets changed, when calling target's _update_inplace.

I think a better solution would be to clear target's cacher in the _update_inplace of the drop method, but I couldn't find a way to do it...